discovery-provider/src/utils/session_manager.py

```diff
@@ -51,6 +69,7 @@ def scoped_session(self, expire_on_commit=True):
         """
         session = self._session_factory()
         session.expire_on_commit = expire_on_commit
+        session.info["src"] = inspect.stack()[2][3]  # get caller's function name
```

Comment:
what if this index doesn't exist? maybe wanna have a try except here

Reply:
Added. In case inspect doesn't give us the caller's name, we'll ignore the error.